docs/user-guide/cross-compilation.md

```diff
@@ -98,6 +104,55 @@ executables you must add package overrides to:
     `integer-gmp`. However, at present, [Haskell.nix][] does not provide
     an option for this.
 
+
+### How to cross-compile your project
+
+Set up your project Haskell package set.
+
+```
+# default.nix
+{ pkgs ? import <nixpkgs> {}
+let
+  # Import the Haskell.nix library,
+  haskell = import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz) {
+    inherit pkgs;
+  };
+
+  # Instantiate a package set using the generated file.
+  pkgSet = haskell.mkCabalProjectPkgSet {
+    plan-pkgs = import ./pkgs.nix;
+    pkg-def-extras = [];
+    modules = [
+      {
+        # You will need to put build fixes here.
+      }
+    ];
+  };
+in
+  pkgSet.config.hsPkgs
+```
+
+Apply that package set to the Nixpkgs cross package sets that you are
+interested in.
+
+```
+# release.nix
+let
+  project = import ./default.nix;
+  pkgs = import <nixpkgs> {};
+  native = import project { inherit pkgs; };
+  crossRasperryPi = import project { pkgs = pkgs.pkgsCross.raspberryPi; };
```

Comment:
Yey, another nixpkgs indirection. Not sure if being explicit in using `import <nixpkgs> { crossSystem = ...; }` would be better. I'm always afraid that those abstractions make it even harder for someone new to nix to understand what is going on :-/

Reply:
Yeah maybe, the `pkgsCross` thing is pretty magic. I have changed it to use `crossSystem` and added an explanation.